llama-index-integrations/vector_stores/llama-index-vector-stores-opensearch/llama_index/vector_stores/opensearch/base.py

```diff
@@ -265,17 +265,34 @@ def _hybrid_search_query(
         k: int,
         filters: Optional[MetadataFilters] = None,
     ) -> Dict:
-        knn_query = self._knn_search_query(
-            embedding_field, query_embedding, k, filters
-        )["query"]
-        lexical_query = {"must": {"match": {text_field: {"query": query_str}}}}
+        knn_query = self._knn_search_query(embedding_field, query_embedding, k, filters)
+        lexical_query = self._lexical_search_query(text_field, query_str, k, filters)
+
+        return {
+            "size": k,
+            "query": {
+                "hybrid": {"queries": [lexical_query["query"], knn_query["query"]]}
+            },
+        }
+
+    def _lexical_search_query(
```

Comment:
does this need to return a dict with field `size`? doesn't look like we end up using this.

Reply:
Hi @nerdai 

Yes they're needed when used to query directly to OpenSearch (outside of hybrid query)
```
        elif query_mode == VectorStoreQueryMode.TEXT_SEARCH:
            search_query = self._lexical_search_query(
                self._text_field, query_str, k, filters=filters
            )
            params = None
```

![Screenshot 2024-06-15 at 08 05 58](https://github.com/run-llama/llama_index/assets/1105460/6325e580-2adf-4c75-9182-d3d9f25df204)

If you check other query generator methods like `_default_approximate_search_query` and  `_default_painless_scripting_query`, they all follow the same pattern
```
    def _default_painless_scripting_query(
        self,
        query_vector: List[float],
        k: int = 4,
        space_type: str = "l2Squared",
        pre_filter: Optional[Union[Dict, List]] = None,
        vector_field: str = "embedding",
    ) -> Dict:
        """For Painless Scripting Search, this is the default query."""
        if not pre_filter:
            pre_filter = MATCH_ALL_QUERY

        source = self.__get_painless_scripting_source(space_type, vector_field)
        return {
            "size": k,
            "query": {
                "script_score": {
                    "query": pre_filter,
                    "script": {
                        "source": source,
                        "params": {
                            "field": vector_field,
                            "query_value": query_vector,
                        },
                    },
                }
            },
        }
```

Reply:
On the hybrid search query, we only took the "query" part from each sub-query because we already defined the size on the higher query.

![Screenshot 2024-06-15 at 08 09 45](https://github.com/run-llama/llama_index/assets/1105460/733d5242-e0a0-43f9-8c15-6613594153d0)

Reply:
Cool! Thanks for the detailed explanation. 😊